NEW : manage several type of dangerous goods for a same parcel

dolibarr95 opened the pull request:
Manage several type of dangerous goods for a same parcel 
Before : dangerous_goods : smallint (so only 1 type can be defined eg 3) 
After dangerous_goods : varchar so can be a list of numbers eg 3,5,8

dolibarr95 committed:
NEW : manage several type of dangerous goods for a same parcel

Manage several type of dangerous goods for a same parcel Before :
dangerous_goods : smallint (so only 1 type can be defined eg 3) After
dangerous_goods : varchar so can be a list of numbers eg 3,5,8

eldy commented on htdocs/install/mysql/tables/llx_expedition_package.sql:
```diff
@@ -12,7 +12,7 @@ create table llx_expedition_package
   weight            float,	       -- weight
   weight_units      integer,	       -- unit of weight
-  dangerous_goods   smallint          DEFAULT 0, -- 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
+  dangerous_goods   varchar(20), --(comma separated list eg : 3,5,7) 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
```
If you change into sql file, you must alos add the migration instruction into the migration file 20.0.0-21.0.0.sql

dolibarr95 replied:
Hello
ok, thanks for the advice I'll do this.

dolibarr95 committed:
Update llx_expedition_package.sql

> Adding a default value to default '0' to keep compatibility seems required.

dolibarr95 committed:
Update 20.0.0-21.0.0.sql

> You must include migration change into same PR so PR is atomic. Include change of #30237

dolibarr95 commented:
Done ✔️

fappels commented on htdocs/install/mysql/tables/llx_expedition_package.sql:
```diff
@@ -12,7 +12,7 @@ create table llx_expedition_package
   weight            float,	       -- weight
   weight_units      integer,	       -- unit of weight
-  dangerous_goods   smallint          DEFAULT 0, -- 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
+  dangerous_goods   varchar(20) DEFAULT '0', --(comma separated list eg : 3,5,7) 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
```
space missing after --

dolibarr95 replied:
tks for your help

dolibarr95 committed:
Update llx_expedition_package.sql

space missing

dolibarr95 commented:
@eldy done ✔️

eldy commented on htdocs/install/mysql/migration/20.0.0-21.0.0.sql:
```diff
@@ -47,3 +47,6 @@ ALTER TABLE llx_contrat ADD COLUMN localtax2 double(24,8) DEFAULT 0;
 ALTER TABLE llx_contrat ADD COLUMN total_ht double(24,8) DEFAULT 0;
 ALTER TABLE llx_contrat ADD COLUMN total_ttc double(24,8) DEFAULT 0;
+
+ALTER TABLE llx_expedition_package CHANGE dangerous_goods dangerous_goods VARCHAR(20) NOT NULL DEFAULT '0';
```
See in head of file the comment
-- To change type of field: ALTER TABLE llx_table MODIFY COLUMN name varchar(60);

dolibarr95 committed:
Update 20.0.0-21.0.0.sql

> -- To change type of field: ALTER TABLE llx_table MODIFY COLUMN name varchar(60);

dolibarr95 commented:
Hello
@eldy and @fappels tks again for your help.
